### PR TITLE
fix(documentation): documentation guidelines

### PR DESCRIPTION
**Pull request by rachelbt**

### CODE REVIEW

File: docs/component-guidline.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+## slots
+
+### slot 1
+lorem ipsum
+
+## CSS variables
+
+## Methods - ? (found in Accordion)
+
+## Accessibility
```
Comment (rinaok) [inline docs/component-guidline.md:63]: Should accessibility be at the end?

File: docs/component-guidline.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+### Slots Name
+Use kebab case for slots name and if containing more than one word should use hyphen.
+
+- `Graphic`
+- `Media` - for slot that are recommended to contain either images or video.
+- `Meta`
+- `Footer`
+- `Content` - should we chang to default?
+-
```
Comment (rinaok) [inline docs/component-guidline.md:32]:
```suggestion
- App Content
```

File: docs/component-guidline.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+## CSS variables
+
+## Methods - ? (found in Accordion)
+
+## Accessibility
+
+## Events - ? (found in banner)
+
+## Use Cases
```
[the remark this answers is not in the document]
Comment (rachelbt) [reply docs/component-guidline.md:67]: yes :)

File: docs/component-guidline.md (hunk 2 above)
Comment (rachelbt) [reply docs/component-guidline.md:32]: I'm not sure app is the right terminology for our components

File: docs/component-guidline.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+- Type: `'member one'` | `'member two'` ...
+- Default: `'default member'`
+
+### Appearance
+Set the `appearance` attribute to change the component's appearance.
+
+### Shape
+Use the `shape` attribute to set the component's edges.
+
+### Density
```
Comment (yinonov) [inline docs/component-guidline.md:81]: we also have size in some components. both are used

File: docs/component-guidline.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
+# Documentation Conventions
+For better and unify components documentation
+
+## Class Names
+
+### Do:
+
+- use `.base` for the first element in the component
+- Use `.control` for form control such as input, buttons ect.
```
Comment (yinonov) [inline docs/component-guidline.md:9]:
```suggestion
- Use `.control` for the main form control such as input, buttons etc'.
```

File: docs/component-guidline.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
+# Documentation Conventions
+For better and unify components documentation
+
+## Class Names
+
+### Do:
```
Comment (yinonov) [inline docs/component-guidline.md:6]: title as _template guide_?

File: docs/component-guidline.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+- use `.base` for the first element in the component
+- Use `.control` for form control such as input, buttons ect.
+- Use kebab case and if containing more than one word should use hyphen.
```
Comment (yinonov) [inline docs/component-guidline.md:10]: kebab case is hypenated

File: docs/component-guidline.md (hunk 6 above)
Comment (yinonov) [inline docs/component-guidline.md:6]: add a point of how to call our class. we have _vivid{Component}_ and _VIVID{Component}_

File: docs/component-guidline.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+### Dont:
+
+- no need for adding vwc to the component name or the css class
+- no need for the component name in the classes
```
Comment (yinonov) [inline docs/component-guidline.md:15]: what?

File: docs/component-guidline.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+- `Heading` for the head or main text in the component.
+- `Sub-Heading`
+- `text`
+
+
+### Slots Name
+Use kebab case for slots name and if containing more than one word should use hyphen.
+
+- `Graphic`
```
Comment (yinonov) [inline docs/component-guidline.md:27]: explain this as well. it sound similar to media

File: docs/component-guidline.md (hunk 2 above)
Comment (yinonov) [reply docs/component-guidline.md:32]: app content is correct only in 2 places I can think of. side-drawer & top-app-bar. these are categorized as templating components

File: docs/component-guidline.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+- 
+
+<hr> 
+
+# Documentation
+## Structure
+
+```
+# Component Name
+One summery line
```
Comment (yinonov) [inline docs/component-guidline.md:41]:
```suggestion
One summary line
```

**Comment by rachelbt** [comment]
> excellent idea!

It was @rinaok idea :)

File: docs/component-guidline.md (hunk 4 above)
Comment (rachelbt) [reply docs/component-guidline.md:81]: size is only in Icons.

File: docs/component-guidline.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+
+### Slots Name
+Use kebab case for slots name and if containing more than one word should use hyphen.
+
+- `Graphic`
+- `Media` - for slot that are recommended to contain either images or video.
+- `Meta`
+- `Footer`
+- `Content` - should we chang to default?
```
[the remark this answers is not in the document]
Comment (rachelbt) [reply docs/component-guidline.md:31]: opened a [separate issue](https://github.com/Vonage/vivid-3/issues/414) for that

File: docs/component-guidline.md [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+
+### Slots Name
+Use kebab case for slots name.
+
+- `Graphic` :for slots that are recommended for decorative element such as custom icon or image
+- `Media` :for slot that are recommended to contain either images or video.
+- `Meta`
+- `Footer`
+- `Content` 
+- `App-content` :for slots in components that categorized as templating components (such as side-drawer, top-app-bar)
```
Comment (yinonov) [inline docs/component-guidline.md:32]:
```suggestion
- `App-content`: for slots in components that categorized as templating components (such as side-drawer, top-app-bar)
```

File: docs/component-guidline.md [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+
+- `Heading` for the head or main text in the component.
+- `Sub-Heading`
+- `text`
+
+
+### Slots Name
+Use kebab case for slots name.
+
+- `Graphic` :for slots that are recommended for decorative element such as custom icon or image
```
Comment (yinonov) [inline docs/component-guidline.md:27]:
```suggestion
- `Graphic` - for slots that are recommended for decorative element such as custom icon or image
```

File: docs/component-guidline.md [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+- `Media` :for slot that are recommended to contain either images or video.
```
Comment (yinonov) [inline docs/component-guidline.md:28]:
```suggestion
- `Media` - for slot that are recommended to contain either images or video.
```

File: docs/component-guidline.md [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+- `Meta`
+- `Footer`
```
Comment (yinonov) [inline docs/component-guidline.md:30]:
```suggestion
- `Footer` / `actions`
```

File: docs/component-guidline.md (hunk 8 above)
Comment (rachelbt) [reply docs/component-guidline.md]: fixed

**Review by rinaok — DISMISSED** [review]

File: docs/component-guidline.md [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+```html preview blocks
+code snipet
+```
+
+#### use of one of these options
+
+- full
+- center
+- blocks
+
```
Comment (yinonov) [inline docs/component-guidline.md:129]:
```suggestion

https://github.com/Vonage/vivid-3/blob/95df74b6284f28cbf4a31279db28d2a40fbc151e/apps/docs/transformers/code-block-demo/layout.js#L7-L11
```

File: docs/component-guidline.md [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+```html preview blocks
+code snipet
+```
+
+#### use of one of these options
+
+- full
+- center
+- blocks
```
Comment (yinonov) [inline docs/component-guidline.md:128]:
```suggestion
- full
- blocks
- columns
- inline (default)
- center
```

File: docs/component-guidline.md [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+
+#### use of one of these options
+
+- full
+- blocks
+- columns
+- inline (default)
+- center
+
+https://github.com/Vonage/vivid-3/blob/95df74b6284f28cbf4a31279db28d2a40fbc151e/apps/docs/transformers/code-block-demo/layout.js#L7-L11
```
Comment (yinonov) [inline docs/component-guidline.md:132]:
```suggestion
```

File: docs/component-guidline.md [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+#### use of one of these options
+
+- full
+- blocks
+- columns
+- inline (default)
+- center
+
```
Comment (yinonov) [inline docs/component-guidline.md:131]:
```suggestion
e.g.
````
```html preview blocks
<!-- content goes here -->
```
````

File: docs/component-guidline.md [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+### Styling wrapper div
+
+- `.cover-fill`
+- `.cover-full`
+- `.demo-gradient`
```
Comment (yinonov) [inline docs/component-guidline.md:136]: I dont think these belong in a docs. the practice of including it should be changed

**Review by yinonov — CHANGES_REQUESTED** [review]

File: docs/component-guidline.md [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+
+### Slot Names
+
+Use kebab case for slots name.
+
+- `graphic` - for slots that are recommended for decorative element such as custom icon or image
+- `media` - for slot that are recommended to contain either images or video.
+- `meta`
+- `footer / actions`
+- `action-ites / action-items` - ?
```
Comment (yinonov) [inline docs/component-guidline.md:32]: ?

File: docs/component-guidline.md [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+- `content`
+- `App-content` - Maybe change it to `app-main` for slots in components that categorized as templating components (such as side-drawer, top-app-bar).
```
Comment (yinonov) [inline docs/component-guidline.md:34]: this apply to only top-bar and side-drawer. let's change to `main` alone?

File: docs/component-guidline.md [hunk 23]
```diff
@@ -0,0 +1,136 @@
...
+- `sub-heading`
+- `text`
+- `label`
+
+### Slot Names
+
+Use kebab case for slots name.
+
+- `graphic` - for slots that are recommended for decorative element such as custom icon or image
+- `media` - for slot that are recommended to contain either images or video.
```
Comment (yinonov) [inline docs/component-guidline.md:29]:
```suggestion
- `media` - for a slot bound to contain visual elements; such as an illustration, video etc'.
```

File: docs/component-guidline.md [hunk 24]
```diff
@@ -0,0 +1,136 @@
...
+- `heading` for the head or main text in the component.
+- `sub-heading`
+- `text`
+- `label`
+
+### Slot Names
+
+Use kebab case for slots name.
+
+- `graphic` - for slots that are recommended for decorative element such as custom icon or image
```
Comment (yinonov) [inline docs/component-guidline.md:28]:
```suggestion
- `graphic` - for a slot bound to contain a decorative figure; such as a custom icon or image
```

File: docs/component-guidline.md [hunk 25]
```diff
@@ -0,0 +1,136 @@
...
+- `media` - for slot that are recommended to contain either images or video.
+- `meta`
+- `footer / actions`
+- `action-ites / action-items` - ?
+- `content`
```
Comment (yinonov) [inline docs/component-guidline.md:33]: remove in favor of main?

File: docs/component-guidline.md [hunk 26]
```diff
@@ -0,0 +1,136 @@
...
+### Structure
+
+```
+# Component Name
+One summary line
+
+- Code import sample
+- Code example preview 
+
+## properties
```
Comment (yinonov) [inline docs/component-guidline.md:49]: consider?
```suggestion
## Members (properties / attributes)
```

File: docs/component-guidline.md [hunk 27]
```diff
@@ -0,0 +1,136 @@
...
+
+### member 1
+lorem ipsum
```
Comment (yinonov) [inline docs/component-guidline.md:52]:
```suggestion

`{{ description }}`

- type: `{{ type }}`
- default: `{{ default }}`
lorem ipsum
```

**Review by yinonov — CHANGES_REQUESTED** [review]

File: docs/component-guidline.md (hunk 1 above)
Comment (rachelbt) [reply docs/component-guidline.md:70]: its not at the end - its the first issue to come after all the API members (properties, slots, css variables)

File: docs/component-guidline.md [hunk 28]
```diff
@@ -0,0 +1,143 @@
+# Documentation Conventions
+
+For better and unify components documentation
+
+## Template Guide
+
+Use the new component generator to create a new component:  
+`npm run nx g @vonage/nx-vivid:component my-component`
+
+### CSS Class Names
```
Comment (yinonov) [inline docs/component-guidline.md:10]: The following headers should be grouped under **Terminology**

**Review by yinonov — CHANGES_REQUESTED** [review]

File: docs/component-guidline.md (hunk 22 above)
Comment (yinonov) [reply docs/component-guidline.md:34]: yeah

File: docs/component-guidline.md (hunk 28 above)
Comment (rachelbt) [reply docs/component-guidline.md:10]: you mean another sub-title? or instead of _CSS Class Names_?

File: docs/component-guidline.md (hunk 20 above)
Comment (yinonov) [reply docs/component-guidline.md:143]: and extracted to code fence attributes (as we did recently)

File: docs/component-guidline.md (hunk 21 above)
Comment (rachelbt) [reply docs/component-guidline.md:32]: typo 😆

**Review by yinonov — APPROVED** [review]

**Pull request merged**